File: sdks/python/reshadx/resources/items.py
"""ReshADX Python SDK - Items Resource"""

from typing import Any, Dict

from ..utils.http import HttpClient


class Items:
    """Items resource"""

    def __init__(self, http: HttpClient):
        self.http = http

    def list(self) -> Dict[str, Any]:
        """Get all items for current user"""
        return self.http.get("/items")

    def get(self, item_id: str) -> Dict[str, Any]:
        """Get specific item details"""
        return self.http.get(f"/items/{item_id}")

    def delete(self, item_id: str) -> Dict[str, bool]:
        """Delete an item (remove bank connection)"""
        return self.http.delete(f"/items/{item_id}")

    def sync(self, item_id: str) -> Dict[str, Any]:
        """Trigger item sync"""
        return self.http.post(f"/items/{item_id}/sync")

    def get_sync_status(self, item_id: str, sync_id: str) -> Dict[str, Any]:
        """Get item sync status"""
        return self.http.get(f"/items/{item_id}/sync/{sync_id}")

    def update_webhook(self, item_id: str, webhook_url: str) -> Dict[str, bool]:
        """Update item webhook URL"""
        data = {"webhookUrl": webhook_url}
        return self.http.patch(f"/items/{item_id}", json_data=data)

File: sdks/typescript/src/resources/items.ts
/**
 * ReshADX TypeScript SDK - Items Resource
 */

import { HttpClient } from '../utils/http';
import { Item } from '../types';

export class Items {
  constructor(private http: HttpClient) {}

  /**
   * Get all items for current user
   */
  async list(): Promise<{ items: Item[] }> {
    return this.http.get('/items');
  }

  /**
   * Get specific item details
   */
  async get(itemId: string): Promise<{ item: Item }> {
    return this.http.get(`/items/${itemId}`);
  }

  /**
   * Delete an item (remove bank connection)
   */
  async delete(itemId: string): Promise<{ success: boolean }> {
    return this.http.delete(`/items/${itemId}`);
  }

  /**
   * Trigger item sync
   */
  async sync(itemId: string): Promise<{
    syncStatus: 'STARTED' | 'IN_PROGRESS' | 'COMPLETED' | 'FAILED';
    syncId: string;
  }> {
    return this.http.post(`/items/${itemId}/sync`);
  }

  /**
   * Get item sync status
   */
  async getSyncStatus(itemId: string, syncId: string): Promise<{
    syncStatus: string;
    progress: number;
    accountsSynced: number;
    transactionsSynced: number;
    error?: string;
  }> {
    return this.http.get(`/items/${itemId}/sync/${syncId}`);
  }

  /**
   * Update item webhook URL
   */
  async updateWebhook(itemId: string, webhookUrl: string): Promise<{ success: boolean }> {
    return this.http.patch(`/items/${itemId}`, { webhookUrl });
  }
}
